### PR TITLE
codeintel: Write SCIP data in `uploads` service

### LINKED ISSUE #45086: RFC 740: Process and persist SCIP uploads
Alter the upload handler to accept SCIP payloads and process them as new data into the schema introduced in the [RFC](https://docs.google.com/document/d/1PuTYmHYLElYKIEW0sSAAUbXf7XtnsMdEq3EshD-O7do/edit).

### LINKED ISSUE #45251: Re-enable `TestConvertLSIF` after fixing flakiness
`TestConvertLSIF` seems flaky because sometimes this symbol has a comment string and sometimes it doesn't: https://github.com/sourcegraph/sourcegraph/blob/c58b9e9ef6116be7412e0047c78d3673315ab634/lib/codeintel/lsif/scip/index_test.go#L79

### PR DESCRIPTION
**Pull request by efritz**
This PR processes and persists SCIP payloads instead of LSIF payloads.

**Do not merge before #45103**.

Fixes #45086. Fixes #45251.

## Test plan

- [main-dry-run](https://buildkite.com/sourcegraph/sourcegraph/builds/187281)

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/internal/background/scip.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	sgtypes "github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/codeintel/pathexistence"
+	"github.com/sourcegraph/sourcegraph/lib/codeintel/precise"
+)
+
+// correlateSCIP reads the content of the given reader as a SCIP index object. The index is processed in
+// the background, and processed documents are emitted on a channel to be persisted to the database.
+//
+// **NOTE TO CONSUMERS OF THIS FUNCTION**:
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/internal/background/scip.go:24]: Note `readPackageAndPackageReferences`

File: enterprise/internal/codeintel/uploads/internal/background/job_worker_handler.go [hunk 2]
```diff
@@ -217,7 +216,7 @@ func (s *handler) HandleRawUpload(ctx context.Context, logger log.Logger, upload
 	}
 
 	return false, withUploadData(ctx, logger, uploadStore, upload.ID, trace, func(r io.Reader) (err error) {
-		groupedBundleData, err := conversion.Correlate(ctx, r, upload.Root, getChildren)
+		correlatedSCIPData, err := correlateSCIP(ctx, r, upload.Root, getChildren)
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/internal/background/job_worker_handler.go:219]: - [ ] First split LSIF and SCIP upload paths; store input type on upload record

Comment (efritz) [reply enterprise/internal/codeintel/uploads/internal/background/job_worker_handler.go:219]: #45129

**Review by Numbers88s — APPROVED** [review]

File: lib/codeintel/lsif/scip/document.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func ConvertLSIFDocument(
 	var (
 		n                           = len(document.Ranges)
 		occurrences                 = make([]*scip.Occurrence, 0, n)
-		documentationBySymbolName   = make(map[string]string, n)
+		documentationBySymbolName   = make(map[string]map[string]struct{}, n)
```
Comment (efritz) [inline lib/codeintel/lsif/scip/document.go:32]: - [ ] Re-enable skipped test: https://github.com/sourcegraph/sourcegraph/issues/45251

**Pull request merged**